File: javascript/guidance/src/docs/select.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
const content =
	"# select\n\nSelects only the specified list of columns into a new table.\n\nExample\n\n| id  | name  | age |\n| --- | ----- | --- |\n| 1   | Bob   | 32  |\n| 2   | Joe   | 35  |\n| 3   | Jenny | 31  |\n\n`select columns['id', 'age']`:\n\n| id  | age |\n| --- | --- |\n| 1   | 32  |\n| 2   | 35  |\n| 3   | 31  |\n"
export default content

File: python/datashaper/datashaper/verbs/select.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Select verb implementation."""

from typing import cast

import pandas as pd
from reactivedataflow import ConfigPort, InputPort, verb

from datashaper import DEFAULT_INPUT_NAME

from .decorators import OutputMode, wrap_verb_result


@verb(
    name="select",
    ports=[
        InputPort(name=DEFAULT_INPUT_NAME, parameter="table", required=True),
        ConfigPort(name="columns", required=True),
    ],
    adapters=[
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def select(table: pd.DataFrame, columns: list[str]) -> pd.DataFrame:
    """Select verb implementation."""
    return cast(pd.DataFrame, table[columns])
